### ISSUE TITLE
Missing data in events collected by RabbitMQ metricbeat module

### ISSUE THREAD

**Issue by jsoriano** [elastic/beats#10014]
Issue created to track actions as follow up of this topic in discuss:
- Discuss Forum URL: https://discuss.elastic.co/t/rabbitmq-plugin-field-problem/162922

Actions:

- [x] Regenerate example `data.json` files as they contain fields that are not created with current code. (https://github.com/elastic/beats/pull/10560)
- [ ] Check support of RabbitMQ 3.6.
- [x] ~~`metricbeat module test` reported to collect an event with empty data, investigate this, seems like some unhandled error.~~

**Comment by immon** [comment]
`metricbeat module test` process [only first event](https://github.com/elastic/beats/blob/v6.5.4/metricbeat/module/rabbitmq/exchange/exchange_test.go#L42) so missing fields it won't be reported in test output.

I verified that if rabbitmq endpoint returns document with `message_stats` the fields like `publish_in.*` and `publish_out.*` are reported correctly.

**Comment by jsoriano** [comment]
Thanks for testing this.

> `metricbeat module test` process [only first event](https://github.com/elastic/beats/blob/v6.5.4/metricbeat/module/rabbitmq/exchange/exchange_test.go#L42) so missing fields it won't be reported in test output.
> 

Yes, but the only event in the discuss topic has an empty name, can a exchange have an empty name?
```
...
      "exchange": {
       "arguments": {},
       "auto_delete": false,
       "durable": true,
       "internal": false,
       "name": "",
       "type": "direct",
       "vhost": "/"
      }
...
```

**Comment by immon** [comment]
According to [rabbitmq tutorial](https://www.rabbitmq.com/tutorials/tutorial-three-python.html) there can be an unnamed exchange.

**Comment by jsoriano** [comment]
Oh, you are right, then this event seems correct, I have removed the point about this in the issue.

**Comment by sayden** [comment]
@jsoriano The description of this issue is a bit vague. After reading the discuss issue few times to understand this one, I think that the description and the problem might be better explained here, please update. 

A missing field in an event is something that is arriving, it's expected in the output but it's not going out. As far as I can see we have some fields that we are not parsing (by design) and some field (`messages`) and subfields that are not arriving so they can't be going out.

> * Regenerate example `data.json` files as they contain fields that are not created with current code.

Can you elaborate this too, please? Those fields are created with the current code by fetching some local files inside the module. I'm not sure if what you are requesting here is that we remove those local files and run the `data.json` generation test with a plain rabbitmq instance. This way we'll only get this values

```json
{
    "@timestamp": "2017-10-12T08:05:34.853Z",
    "agent": {
        "hostname": "host.example.com",
        "name": "host.example.com"
    },
    "event": {
        "dataset": "rabbitmq.exchange",
        "duration": 115000,
        "module": "rabbitmq"
    },
    "metricset": {
        "name": "exchange"
    },
    "rabbitmq": {
        "exchange": {
            "arguments": {},
            "auto_delete": false,
            "durable": true,
            "internal": false,
            "name": "",
            "type": "direct"
        },
        "vhost": "/"
    },
    "service": {
        "address": "172.17.0.2:15672",
        "type": "rabbitmq"
    }
}
```

Is this what we want? In such case, why? Isn't it better to update the module to ensure that the testing rabbitmq instance return an entire set of values that are supported in the Metricset?

**Comment by jsoriano** [comment]
@sayden regarding the pending issues here:

* Regenerate the `data.json` with actual data from a server. When the issue was created it contained fields not documented. Now it seems that it is obtained from an example input (after #10560). Maybe it is ok now, but I'd still check that the exported data with this input is similar to the exported from a real server.
* Check support with RabbitMQ 3.6. Currently we support 3.7, it'd be to check that we also support older versions that people is still using. It'd be enough with checking that it actually works. It'd be also nice to run the tests against both versions as we do with some other services.

>  @ruflin @jsoriano are we ok by adding those fields?

I am ok with that if there is some scenario in which these values are not always zero.

**Comment by sayden** [comment]
@jsoriano we should review this issue to close it I think. If we want to do more things around it maybe we should open a new issue with a accurate description about a single task (single PR) to tackel. WDYT? :)
